src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    if (this[0] !== null && this[0] !== undefined) {
+      resultString = `${this[0]}`;
```

Comment:
You don't need to use back quotes here

Reply:
Also, do you really need to do this here for the first element? You have the same condition(`this[i] !== null && this[i] !== undefined`) inside the loop. Try to combine them(but need to change some logic), your code will be more clear